libavcodec/vaapi_encode.c

```diff
@@ -1845,6 +1845,30 @@ static av_cold int vaapi_encode_init_gop_structure(AVCodecContext *avctx)
         ref_l1 = attr.value >> 16 & 0xffff;
     }
 
+#if VA_CHECK_VERSION(1, 9, 0)
+    // if backend doesn't support P frame, convert P to forward-prediction B.
+    // if backend doesn't support future prediction, convert B to forward-prediction B.
+    if (!ctx->b_frame_strategy) {
+        attr = (VAConfigAttrib) { VAConfigAttribPredictionDirection };
+        vas = vaGetConfigAttributes(ctx->hwctx->display,
+                                    ctx->va_profile,
+                                    ctx->va_entrypoint,
+                                    &attr, 1);
+        if (vas != VA_STATUS_SUCCESS) {
+            av_log(avctx, AV_LOG_ERROR, "Failed to query prediction direction "
+                   "attribute: %d (%s).\n", vas, vaErrorStr(vas));
+            return AVERROR_EXTERNAL;
+        }
```

Comment:
It is taken as a warning instead of an error when `vaGetConfigAttributes` doesn't return  VA_STATUS_SUCCESS in 3c68549f5dca294bb415ae4c2da667db6359f306, and then it check whether attr.value == VA_ATTRIB_NOT_SUPPORTED. I think we should keep the logic consistent.

Reply:
Will fix in next version.

Reply:
Will change to WARNING in next version.
Thanks.